soc/nxp/imxrt/imxrt7xx/cm33/soc.c

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright 2024 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief System/hardware module for NXP RT5XX platform
+ *
+ * This module provides routines to initialize and support board-level
+ * hardware for the RT5XX platforms.
+ */
+
+#include <zephyr/init.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/linker/sections.h>
+#include <soc.h>
+#include "fsl_power.h"
+#include "fsl_clock.h"
+
+#ifdef CONFIG_HAS_MCUX_CACHE
+#include <fsl_cache.h>
+#endif
+
+#ifdef CONFIG_FLASH_MCUX_XSPI_XIP
+#include "flash_clock_setup.h"
+#endif
+
+#define BOARD_SYSOSC_SETTLING_US 220U   /*!< Board System oscillator settling time in us */
+#define BOARD_XTAL32K_CLK_HZ 32768U     /*!< Board xtal32K frequency in Hz */
+#define BOARD_XTAL_SYS_CLK_HZ 24000000U /*!< Board xtal frequency in Hz */
+
+const clock_main_pll_config_t g_mainPllConfig_BOARD_BootClockRUN = {
+	.main_pll_src  = kCLOCK_MainPllOscClk, /* OSC clock */
+	.numerator     = 0, /* Numerator of the SYSPLL0 fractional loop divider is 0 */
+	.denominator   = 1, /* Denominator of the SYSPLL0 fractional loop divider is 1 */
+	.main_pll_mult = kCLOCK_MainPllMult22  /* Divide by 22 */
+};
+
+const clock_audio_pll_config_t g_audioPllConfig_BOARD_BootClockRUN = {
+	.audio_pll_src  = kCLOCK_AudioPllOscClk,   /* OSC clock */
+	.numerator      = 5040,  /* Numerator of the Audio PLL fractional loop divider is 0 */
+	.denominator    = 27000, /* Denominator of the Audio PLL fractional loop divider is 1 */
+	.audio_pll_mult = kCLOCK_AudioPllMult22,   /* Divide by 22 */
+	.enableVcoOut   = true
+};
+
+/*FUNCTION**********************************************************************
+ *
+ * Function Name : BOARD_XspiClockSafeConfig
+ * Description   : XSPI clock source safe configuration weak function.
+ *                 Called before clock source(Such as PLL, Main clock) configuration.
+ * Note          : Users need override this function to change XSPI clock source to stable
+ *                 source when executing code on XSPI memory(XIP). If XIP, the function
+ *                 should runs in RAM and move the XSPI clock source to an stable clock to
+ *                 avoid instruction/data fetch issue during clock updating.
+ *END**************************************************************************/
+__attribute__((weak)) void BOARD_XspiClockSafeConfig(void)
+{
+}
+
+__attribute__((weak)) void BOARD_SetXspiClock(XSPI_Type *base, uint32_t src, uint32_t divider)
+{
+}
+
+void z_arm_platform_init(void)
+{
+	SystemInit();
+}
+
+static int imxrt7xx_soc_init(void)
+{
+#ifdef CONFIG_SOC_MIMXRT798S_CPU0
```

Comment:
if() wherever possible for whole PR (sometimes, you have to use #if , but usually not)

Reply:
Same like above.